README.md
[![Build Status](https://travis-ci.org/mudler/Mojolicious-Plugin-Angular-MaterialDesign.svg?branch=master)](https://travis-ci.org/mudler/Mojolicious-Plugin-Angular-MaterialDesign)
# NAME

Mojolicious::Plugin::Angular::MaterialDesign - Mojolicious + https://material.angularjs.org/

# DESCRIPTION

[Mojolicious::Plugin::Angular::MaterialDesign](https://metacpan.org/pod/Mojolicious::Plugin::Angular::MaterialDesign) is used to include [https://material.angularjs.org/](https://material.angularjs.org/)
CSS and JavaScript files into your project.

This is done with the help of [Mojolicious::Plugin::AssetPack](https://metacpan.org/pod/Mojolicious::Plugin::AssetPack).

# SYNOPSIS

## Mojolicious

    use Mojolicious::Lite;
    plugin "Angular::MaterialDesign";
    get "/" => "index";
    app->start;

## Template

    <!doctype html>
    <html>
      <head>
        % # ... your angular asset must be loaded before
        %= asset "materialdesign.css"
        %= asset "materialdesign.js"
      </head>
      <body>
        <p class="alert alert-danger">Danger, danger! High Voltage!</p>
      </body>
    </html>

TIP! You might want to load [Mojolicious::Plugin::AssetPack](https://metacpan.org/pod/Mojolicious::Plugin::AssetPack) yourself to specify
options.

# METHODS

## asset\_path

    $path = Mojolicious::Plugin::Angular::MaterialDesign->asset_path();
    $path = $self->asset_path();

Returns the base path to the assets bundled with this module.

## register

    $app->plugin("Angular::MaterialDesign");

Loads the plugin and register the static paths that includes the css and js.

# CREDITS

[angular/material](https://github.com/angular/material) <contributors|https://github.com/angular/material/graphs/contributors>

# LICENSE

Copyright (C) mudler.

This library is free software; you can redistribute it and/or modify
it under the same terms as Perl itself.

# AUTHOR

mudler <mudler@dark-lab.net>


Changes
Revision history for Perl extension Mojolicious-Plugin-Angular-MaterialDesign

{{$NEXT}}

1.1.0 2016-08-28T11:00:12Z

    - Updating Material component and fixing missing dependency

0.603 2014-12-28T10:01:56Z

    - minor fixes

0.602 2014-12-28T09:58:13Z

    - cleanup

0.601 2014-12-27T13:04:09Z

    - minor fixes

0.6 2014-12-27T12:59:24Z

    - first release



Makefile.PL
# =========================================================================
# THIS FILE IS AUTOMATICALLY GENERATED BY MINILLA.
# DO NOT EDIT DIRECTLY.
# =========================================================================

use 5.008_001;
use strict;

use ExtUtils::MakeMaker;


use File::Copy;

print "cp META.json MYMETA.json\n";
copy("META.json","MYMETA.json") or die "Copy failed(META.json): $!";

if (-f 'META.yml') {
    print "cp META.yml MYMETA.yml\n";
    copy("META.yml","MYMETA.yml") or die "Copy failed(META.yml): $!";
} else {
    print "There is no META.yml... You may install this module from the repository...\n";
}

my %args;
if ($ExtUtils::MakeMaker::VERSION >= 6.64) {
    # *_REQUIRES was supported.

    $args{CONFIGURE_REQUIRES} = {"ExtUtils::MakeMaker" => "6.64"};
    $args{BUILD_REQUIRES}     = {};
    $args{TEST_REQUIRES}      = {"Mojolicious" => 0,"Test::Mojo" => 0,"Test::More" => "0.98"};
    $args{PREREQ_PM}          = {"Mojo::Base" => 0,"Mojolicious::Plugin::AssetPack" => 0,"perl" => "5.008001"};
} else {
    $args{PREREQ_PM}          = {"ExtUtils::MakeMaker" => "6.64","Mojo::Base" => 0,"Mojolicious" => 0,"Mojolicious::Plugin::AssetPack" => 0,"Test::Mojo" => 0,"Test::More" => "0.98","perl" => "5.008001"};
}

if ($ExtUtils::MakeMaker::VERSION >= 6.57_01) {
    $args{NO_MYMETA} = 1;
}

WriteMakefile(
    NAME     => 'Mojolicious::Plugin::Angular::MaterialDesign',
    DISTNAME => 'Mojolicious-Plugin-Angular-MaterialDesign',
    VERSION  => '0.603',
    %args,
);


lib/Mojolicious/Plugin/Angular/MaterialDesign.pm
package Mojolicious::Plugin::Angular::MaterialDesign;
use 5.008001;
use strict;
use warnings;

use Mojo::Base 'Mojolicious::Plugin';
use Mojolicious::Plugin::AssetPack;
use File::Spec::Functions 'catdir';
use Cwd ();

our $VERSION = '0.603';

my @DEFAULT_CSS_FILES = qw( angular-material.min.css );
my @DEFAULT_JS_FILES  = qw( angular-material.min.js);

sub asset_path {
    my ( $class ) = @_;
    my $path = Cwd::abs_path(__FILE__);
    $path =~ s!\.pm$!!;
    return $path;
}

sub register {
    my ( $self, $app, $config ) = @_;

    $app->plugin('AssetPack') unless eval { $app->asset };

    $config->{css} ||= [@DEFAULT_CSS_FILES];
    $config->{js}  ||= [@DEFAULT_JS_FILES];
    $config->{jquery} //= 1;

    push @{ $app->static->paths }, $self->asset_path;

    # TODO: 'bootstrap_resources.scss'
    if ( @{ $config->{css} } ) {
        $app->asset( 'materialdesign.css' => map {"/css/$_"}
                @{ $config->{css} } );
    }

    if ( @{ $config->{js} } ) {
        $app->asset( 'materialdesign.js' => map {"/js/$_"} @{ $config->{js} },
        );
    }

}

1;
__END__

=encoding utf-8

=head1 NAME

Mojolicious::Plugin::Angular::MaterialDesign - Mojolicious + https://material.angularjs.org/

=head1 DESCRIPTION

L<Mojolicious::Plugin::Angular::MaterialDesign> is used to include L<https://material.angularjs.org/>
CSS and JavaScript files into your project.

This is done with the help of L<Mojolicious::Plugin::AssetPack>.

=head1 SYNOPSIS

=head2 Mojolicious

  use Mojolicious::Lite;
  plugin "Angular::MaterialDesign";
  get "/" => "index";
  app->start;

=head2 Template

  <!doctype html>
  <html>
    <head>
      % # ... your angular asset must be loaded before
      %= asset "materialdesign.css"
      %= asset "materialdesign.js"
    </head>
    <body>
      <p class="alert alert-danger">Danger, danger! High Voltage!</p>
    </body>
  </html>

TIP! You might want to load L<Mojolicious::Plugin::AssetPack> yourself to specify
options.


=head1 METHODS

=head2 asset_path

  $path = Mojolicious::Plugin::Angular::MaterialDesign->asset_path();
  $path = $self->asset_path();

Returns the base path to the assets bundled with this module.

=head2 register

  $app->plugin("Angular::MaterialDesign");

Loads the plugin and register the static paths that includes the css and js.

=head1 CREDITS

L<angular/material|https://github.com/angular/material> L<contributors|https://github.com/angular/material/graphs/contributors>

=head1 LICENSE

Copyright (C) mudler.

This library is free software; you can redistribute it and/or modify
it under the same terms as Perl itself.

=head1 AUTHOR

mudler E<lt>mudler@dark-lab.netE<gt>

=cut

